File: api/list.py
import os
import json
import psycopg2
from http.server import BaseHTTPRequestHandler

class handler(BaseHTTPRequestHandler):
    def _cors(self):
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Access-Control-Allow-Headers', 'Content-Type')
        self.send_header('Access-Control-Allow-Methods', 'GET,POST,OPTIONS')

    def do_OPTIONS(self):
        self.send_response(204)
        self._cors()
        self.end_headers()

    def do_GET(self):
        try:
            conn = psycopg2.connect(os.environ.get('DATABASE_URL'))
            cur = conn.cursor()
            cur.execute('SELECT data FROM registrations ORDER BY id DESC')
            rows = [r[0] for r in cur.fetchall() if r and r[0]]
            cur.close(); conn.close()
            self.send_response(200)
            self._cors()
            self.send_header('Content-Type', 'application/json')
            self.end_headers()
            self.wfile.write(json.dumps({'success': True, 'rows': rows}).encode('utf-8'))
        except Exception as e:
            self.send_response(500)
            self._cors()
            self.send_header('Content-Type', 'application/json')
            self.end_headers()
            self.wfile.write(json.dumps({'success': False, 'error': str(e)}).encode('utf-8'))

File: netlify/functions/list.js
const { getStore } = require('@netlify/blobs')

exports.handler = async (event) => {
  const headers = {
    'Access-Control-Allow-Origin': '*',
    'Access-Control-Allow-Headers': 'Content-Type',
    'Access-Control-Allow-Methods': 'OPTIONS,GET,DELETE',
    'Content-Type': 'application/json'
  }
  if (event.httpMethod === 'OPTIONS') return { statusCode: 204, headers }
  try {
    const store = getStore('registrations')
    if (event.httpMethod === 'DELETE') {
      let idxRaw = await store.get('index')
      let idx = []
      if (idxRaw) { try { idx = JSON.parse(idxRaw) } catch (_) { idx = [] } }
      for (const item of idx) { try { await store.delete(item.key) } catch (_) {} }
      await store.delete('index')
      return { statusCode: 200, headers, body: JSON.stringify({ success: true }) }
    }
    let idxRaw = await store.get('index')
    let idx = []
    if (idxRaw) { try { idx = JSON.parse(idxRaw) } catch (_) { idx = [] } }
    const rows = []
    for (const item of idx) {
      const rRaw = await store.get(item.key)
      if (rRaw) { try { rows.push(JSON.parse(rRaw)) } catch (_) {} }
    }
    return { statusCode: 200, headers, body: JSON.stringify({ success: true, rows }) }
  } catch (e) {
    return { statusCode: 500, headers, body: JSON.stringify({ success: false, message: 'Server error' }) }
  }
}
